back/infolica/views/balance.py

```diff
@@ -168,56 +151,77 @@ def balance_from_file_view(request):
     if not check_connected(request):
         raise exc.HTTPForbidden()
 
-    input_file = request.params["filepath"] if 'filepath' in request.params else None
+    input_file = request.params["filepath"] if "filepath" in request.params else None
+    affaire_id = request.params["affaire_id"] if "affaire_id" in request.params else None
 
-   
-    # Open balance file and get table of balance
+    ## Open balance file and get cadastre and table of balance
     doc = Document(input_file)
+
+    # get cadastre from affaire
+    cadastre = request.dbsession.query(Cadastre).join(Affaire, Affaire.cadastre_id == Cadastre.id).filter(Affaire.id == affaire_id).first()
+    # get cadastre from balance file if exists
+    for p in doc.sections[0].header.paragraphs:
+        text = p.text.lower()
+        if text.startswith("cadastre"):
+            test = re.match("(cadastre (de|du) )", text.lower())
+
+            cadastre_ = text[test.span()[1] :]
+
+            cadastre = request.dbsession.query(
+                Cadastre,
+                func.similarity(Cadastre.nom, cadastre_).label("similarity")
```

Comment:
Isn't there a better (or more deterministic) way to get the cadastre?

Reply:
It could be great, but I haven't found a better solution till now. I check it once more 👍

Reply:
At the moment, I haven't found a better solution...